--- Makefile.org	2014-02-14 01:57:37.000000000 -0800
+++ Makefile	2014-02-14 06:48:57.000000000 -0800
@@ -23,20 +23,11 @@
 
 LDFLAGS += -shared
 
-PREFIX ?= /usr
+PREFIX = $(prefix)
 DESTDIR ?=
 INCDIR = $(DESTDIR)$(PREFIX)/include
 
 LIBDIR = $(DESTDIR)$(PREFIX)/lib
-# on x86_64, we might have /usr/lib64 directory instead of /usr/lib
-MACHINE := $(shell uname -m)
-ifeq ($(MACHINE), x86_64)
-ifeq (,$(wildcard $(LIBDIR)))
-LIBDIR = $(DESTDIR)$(PREFIX)/lib64
-else
-LIBDIR = $(DESTDIR)$(PREFIX)/lib
-endif
-endif
 
 INSTALL_BIN ?= install
 INSTALL_DATA ?= $(INSTALL_BIN) -m0644
