--- make.sh.org	2014-02-14 01:14:07.000000000 -0800
+++ make.sh	2014-02-14 01:34:36.000000000 -0800
@@ -42,16 +42,4 @@
 export PREFIX=/usr/local
 fi
 
-case "$1" in
-  "" ) build;;
-  "default" ) build;;
-  "install" ) install;;
-  "nix32" ) CFLAGS=-m32 LDFLAGS=-m32 build;;
-  "cross-win32" ) CROSS=i686-w64-mingw32- build;;
-  "cross-win64" ) CROSS=x86_64-w64-mingw32- build;;
-  "cygwin-mingw32" ) CROSS=i686-pc-mingw32- build;;
-  "cygwin-mingw64" ) CROSS=x86_64-w64-mingw32- build;;
-  "clang" ) CC=clang build;;
-  "gcc" ) CC=gcc build;;
-  * ) echo "Usage: make.sh [nix32|cross-win32|cross-win64|cygwin-mingw32|cygwin-mingw64|clang|gcc]"; exit 1;;
-esac
+build
